### PR TITLE
chore: Bump datastore-s3 to 5.0.0

### PR DESCRIPTION
**Pull request by ukstv**

### CODE REVIEW

**Comment by valmack** [comment]
Before this goes in we need to double check which repos are sharded already. I don't think this will work if we start sharding something that previously wasn't. Will report back shortly

File: packages/ipfs-daemon/src/s3-lock.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
+import path from 'path';
+import S3Store from 'datastore-s3';
+
+/**
+ * Uses an object in an S3 bucket as a lock to signal that an IPFS repo is in use.
+ * When the object exists, the repo is in use. You would normally use this to make
+ * sure multiple IPFS nodes don’t use the same S3 bucket as a datastore at the same time.
+ */
+export class S3Lock {
+  readonly s3: S3Store;
```
Comment (valmack) [inline packages/ipfs-daemon/src/s3-lock.ts:10]: Why do we need this S3 lock, given we have been operating without it?

File: packages/ipfs-daemon/src/create-repo.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        ...storeConfig,
+        sharding: false,
+      },
+      datastore: {
+        ...storeConfig,
+        sharding: false,
+      },
+      pins: {
+        ...storeConfig,
+        sharding: true,
```
Comment (valmack) [inline packages/ipfs-daemon/src/create-repo.ts:74]: Not positive this was previously sharded

**Review by valmack — COMMENTED** [review]
Will check about sharding (after I feed the cat 😄)

File: packages/ipfs-daemon/src/create-repo.ts (hunk 2 above)
Comment (ukstv) [reply packages/ipfs-daemon/src/create-repo.ts:74]: It was not.

File: packages/ipfs-daemon/src/create-repo.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+      keys: S3Store,
+      datastore: S3Store,
+      pins: S3Store,
+    },
+    storageBackendOptions: {
+      root: {
+        ...storeConfig,
+      },
+      blocks: {
+        ...storeConfig,
```
Comment (valmack) [inline packages/ipfs-daemon/src/create-repo.ts:60]: We actually do want sharding here for blocks, as that is what we currently have.

**Review by valmack — COMMENTED** [review]
Great just need to add sharding back to blocks. Confirmed that this is the only store that is sharded in our buckets currently

File: packages/ipfs-daemon/src/create-repo.ts (hunk 3 above)
Comment (ukstv) [reply packages/ipfs-daemon/src/create-repo.ts:60]: okay.

Comment (ukstv) [reply packages/ipfs-daemon/src/create-repo.ts:60]: done.

Comment (ukstv) [reply packages/ipfs-daemon/src/create-repo.ts:60]: heck, committed in a different branch

Comment (ukstv) [reply packages/ipfs-daemon/src/create-repo.ts:60]: now it is back.

**Review by valmack — APPROVED** [review]
👍

**Pull request merged**